server/services/group.js

```diff
@@ -129,13 +138,63 @@ function scheduleEPRotation(group) {
 }
 
 function rotateEscalationPolicy(group) {
-  let subscribers = group.escalationPolicy.subscribers;
+  const subscribers = group.escalationPolicy.subscribers;
   subscribers.push(subscribers.shift());
-  subscribers = subscribers.map(s => s.toString());
   const groupUpdates = { lastRotated: new Date() };
   return scheduleEPRotation(group)
     .then(() => updateGroup(group.name, groupUpdates))
-    .then(() => updateEscalationPolicy(group.name, { subscribers }));
+    .then(() => updateEscalationPolicy(group.name, { subscribers: morphSubscribers(subscribers) }));
+}
+
+function scheduleDeactivateUser(group, userId, deactivateDate, reactivateDate) {
+  const subscribers = group.escalationPolicy.subscribers;
+  const subscriberToOverride = _.find(subscribers, s => s.userId.toString() === userId.toString());
+  subscriberToOverride.deactivateDate = deactivateDate;
+  subscriberToOverride.reactivateDate = reactivateDate;
+
+  return updateEscalationPolicy(group.name, { subscribers: morphSubscribers(subscribers) })
+    .then((updatedGroup) => {
+      if (deactivateDate <= new Date()) {
+        return deactivateUser(updatedGroup, userId, reactivateDate);
+      }
+      scheduler.scheduleJob(deactivateDate,
+                            deactivateUser.bind(null, updatedGroup, userId, reactivateDate));
+      return updatedGroup;
+    })
+    .then(gWithDeactivatedUser => Promise.resolve({ deactivateDate, group: gWithDeactivatedUser }));
+}
+
+function scheduleReactivateUser(group, userId) {
+  const subscribers = group.escalationPolicy.subscribers;
+  const toReactivate = _.find(subscribers, s => s.userId.toString() === userId.toString());
+  const reactivateDate = toReactivate.reactivateDate;
+
+  if (reactivateDate <= new Date()) {
+    return reactivateUser(group, userId)
+      .then(updatedGroup => Promise.resolve({ reactivateDate, group: updatedGroup }));
+  }
+  scheduler.scheduleJob(reactivateDate,
+                        reactivateUser.bind(null, group, userId));
```

Comment:
This binds the group and userId to the reactiveUser so you can use it later, right?